tests/top_bar_component.test.ts

```diff
@@ -933,13 +938,138 @@ test("Clicking on a topbar button triggers two renders", async () => {
   jest.advanceTimersByTime(DEBOUNCE_TIME + 10); // wait for the debounce of session.move
   jest.useRealTimers();
 
-  const triggerRender = jest.fn();
-  model.on("update", {}, triggerRender);
-  env["__spreadsheet_stores__"].on("store-updated", null, triggerRender);
+  const modelRender = jest.fn();
+  const storeRender = jest.fn();
+  model.on("update", {}, modelRender);
+  env["__spreadsheet_stores__"].on("store-updated", null, storeRender);
 
   await click(fixture, ".o-spreadsheet-topbar [title='Bold (Ctrl+B)']");
 
-  // one render from the collaboration session
+  // two renders from the collaboration session
   // one from the top bar interaction
-  expect(triggerRender).toHaveBeenCalledTimes(2);
+  expect(modelRender).toHaveBeenCalledTimes(2);
+  expect(storeRender).toHaveBeenCalledTimes(1);
+});
+
+describe("Responsive Top bar behaviour", () => {
+  afterEach(() => {
+    spreadsheetWidth = 1000;
+  });
+
+  describe("items are hidden when the screen is resized", () => {
+    const widthThresholds = topbarToolsWidthThresholds.map((threshold, index) => [
+      threshold,
+      index,
+    ]);
+    const categories = topBarToolBarRegistry.getCategories();
+
+    test.each(widthThresholds)("Screen slightly smaller than %spx ", async (threshold, index) => {
+      spreadsheetWidth = threshold - 1;
+      await mountParent();
+      await nextTick();
+      expect(fixture.querySelectorAll(".o-toolbar-tools>div")).toHaveLength(
+        categories.length - index
+      );
+    });
+
+    test("toolbar items hidden are moved in a popover", async () => {
+      await mountParent();
+      await click(fixture, '.o-menu-item-button[title="Vertical align"]');
+
+      expect(fixture.querySelector('.o-menu-item-button[title="Vertical align"]')).not.toBeNull();
+      expect(fixture.querySelector('.o-menu-item-button[title="Horizontal align"]')).not.toBeNull();
+      expect(fixture.querySelector('.o-menu-item-button[title="Wrapping"]')).not.toBeNull();
+
+      spreadsheetWidth = 690; // hides the last 2 categories
+      window.resizers.resize();
+      await nextTick();
+
+      expect(fixture.querySelector('.o-menu-item-button[title="Vertical align"]')).toBeNull();
+      expect(fixture.querySelector('.o-menu-item-button[title="Horizontal align"]')).toBeNull();
+      expect(fixture.querySelector('.o-menu-item-button[title="Wrapping"]')).toBeNull();
+
+      await click(fixture, "#moreTools"); // open the popover
```

Comment:
is it the first time we use actual ids in the HTML ? :sweat_smile:

Reply:
aaaaand it's gone